android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java

```diff
@@ -0,0 +1,1646 @@
+                mPopupView.findViewById(R.id.auto_contribute_summary_layout)
+                        .setVisibility(walletType.equals(BraveWalletProvider.UPHOLD) ? View.VISIBLE
+                                                                                     : View.GONE);
+            }
+        }
+
+        mBraveRewardsNativeWorker.GetPendingContributionsTotal();
+    }
+
+    @Override
+    public void OnGetAdsAccountStatement(boolean success, double nextPaymentDate,
+            int adsReceivedThisMonth, double earningsThisMonth, double earningsLastMonth) {
+        mAdsStatementLayout.setAlpha(1.0f);
+        mAdsStatementProgress.setVisibility(View.GONE);
+        DecimalFormat df = new DecimalFormat("#.###");
+        df.setRoundingMode(RoundingMode.FLOOR);
+        df.setMinimumFractionDigits(3);
+        TextView batBalanceAdsText = mPopupView.findViewById(R.id.bat_balance_ads_text);
+        batBalanceAdsText.setText(df.format(earningsThisMonth));
+        double usdValue = earningsThisMonth * mBraveRewardsNativeWorker.GetWalletRate();
+        String usdText =
+                String.format(mPopupView.getResources().getString(R.string.brave_ads_statement_usd),
+                        String.format(Locale.getDefault(), "%.2f", usdValue));
+        TextView usdBalanceAdsText = mPopupView.findViewById(R.id.usd_balance_ads_text);
+        usdBalanceAdsText.setText(usdText);
+    }
+
+    @Override
+    public void OnRewardsParameters(int errorCode) {
+        mWalletBalanceLayout.setAlpha(1.0f);
+        mWalletBalanceProgress.setVisibility(View.GONE);
+        if (errorCode == BraveRewardsNativeWorker.LEDGER_OK) {
+            if (mBraveRewardsNativeWorker != null) {
+                BraveRewardsBalance walletBalanceObject =
+                        mBraveRewardsNativeWorker.GetWalletBalance();
+                double walletBalance = 0;
+                if (walletBalanceObject != null) {
+                    walletBalance = walletBalanceObject.getTotal();
+                }
+
+                DecimalFormat df = new DecimalFormat("#.###");
+                df.setRoundingMode(RoundingMode.FLOOR);
+                df.setMinimumFractionDigits(3);
+                TextView batBalanceText = mPopupView.findViewById(R.id.bat_balance_text);
+                batBalanceText.setText(df.format(walletBalance));
+                double usdValue = walletBalance * mBraveRewardsNativeWorker.GetWalletRate();
+                String usdText =
+                        String.format(mPopupView.getResources().getString(R.string.brave_ui_usd),
+                                String.format(Locale.getDefault(), "%.2f", usdValue));
+                TextView usdBalanceText = mPopupView.findViewById(R.id.usd_balance_text);
+                usdBalanceText.setText(usdText);
+            }
+        } else if (errorCode == BraveRewardsNativeWorker.LEDGER_ERROR) { // No Internet connection
+            String args[] = {};
+            Log.e(TAG, "LEDGER_ERROR");
```

Comment:
Maybe we could say something like `Failed to fetch rewards parameters from server` (or something like that) here?